services/ymax-planner/test/kv-store-flow.test.ts

```diff
@@ -0,0 +1,214 @@
+/**
+ * Tests for the functionality of flows involving the KV store
+ */
+import test from 'ava';
+
+import { TxType } from '@aglocal/portfolio-contract/src/resolver/constants.js';
+import { createMockPendingTxData } from '@aglocal/portfolio-contract/tools/mocks.ts';
+import { getTxBlockLowerBound, setTxBlockLowerBound } from '../src/kv-store.ts';
+import { handlePendingTx } from '../src/pending-tx-manager.ts';
+import { EVENTS } from '../src/watchers/gmp-watcher.ts';
+import {
+  createMockGmpExecutionEvent,
+  createMockPendingTxOpts,
+} from './mocks.ts';
+
+const setupEnvironment = ({
+  latestBlock,
+  blockWithEvent,
+  txId,
+  contractAddress,
+}) => {
+  const logs: string[] = [];
+  const mockLog = (...args: unknown[]) => logs.push(args.join(' '));
+
+  const destinationAddress: `${string}:${string}:${string}` = `eip155:42161:${contractAddress}`;
+
+  const gmpTx = createMockPendingTxData({
+    type: TxType.GMP,
+    destinationAddress,
+  });
+
+  const chainId = 'eip155:42161';
+  const opts = createMockPendingTxOpts();
+  const mockProvider = opts.evmProviders[chainId] as any;
```

Comment:
This cast to `any` is a liability. We should be explicit about our deviations.
```suggestion
  const mockProvider = opts.evmProviders[chainId];
```